### ISSUE TITLE
JSX whitespace incorrect in props

### ISSUE THREAD

**Issue by lencioni** [babel/babel#1276]
Given the following JSX:

```jsx
var HelloMessage = React.createClass({
  render: function() {
    return <div>Hello {this.props.name}</div>;
  }
});

React.render(<HelloMessage name={
  <span>
    Sebastian
  </span>
} />, mountNode);
```

Facebook's transformer (0.13.1) changes this to:

```js
var HelloMessage = React.createClass({displayName: "HelloMessage",
  render: function() {
    return React.createElement("div", null, "Hello ", this.props.name);
  }
});

React.render(React.createElement(HelloMessage, {name: 
  React.createElement("span", null, 
    "Sebastian"
  )
}), mountNode);
```

But Babel's transformer changes it to:

```js
"use strict";

var HelloMessage = React.createClass({
  displayName: "HelloMessage",

  render: function render() {
    return React.createElement(
      "div",
      null,
      "Hello ",
      this.props.name
    );
  }
});

React.render(React.createElement(HelloMessage, { name: React.createElement(
    "span",
    null,
    "\n    Sebastian\n  "
  ) }), mountNode);
```

[See it in the REPL](https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&playground=true&code=var%20HelloMessage%20%3D%20React.createClass(%7B%0A%20%20render%3A%20function()%20%7B%0A%20%20%20%20return%20%3Cdiv%3EHello%20%7Bthis.props.name%7D%3C%2Fdiv%3E%3B%0A%20%20%7D%0A%7D)%3B%0A%0AReact.render(%3CHelloMessage%20name%3D%7B%0A%20%20%3Cspan%3E%0A%20%20%20%20Sebastian%0A%20%20%3C%2Fspan%3E%0A%7D%20%2F%3E%2C%20mountNode)%3B)

The newlines and extra spaces in the name prop in the Babel example are problematic. Tested on whatever version is currently powering the REPL.

**Comment by sebmck** [comment]
This was a weird case. The children were being turned into strings but they weren't being properly sanitised/cleaned up because the root node wasn't being visited.

**Comment by sebmck** [comment]
Fixed as of 5.1.11, thanks!

**Comment by lencioni** [comment]
Thanks for the fast response!
